SharedPackages/BrowserServicesKit/Sources/SecureStorage/SecureStorageDatabaseProvider.swift

```diff
@@ -89,20 +88,19 @@ open class GRDBSecureStorageDatabaseProvider: SecureStorageDatabaseProvider {
         return writer
     }
 
-    private static func recreateDatabase(withKey key: Data, databaseURL: URL) throws {
+    private static func recreateDatabase(withKey key: Data,
+                                         databaseURL: URL,
+                                         writerType: DatabaseWriterType,
+                                         registerMigrationsHandler: (inout DatabaseMigrator) throws -> Void) throws -> DatabaseWriter {
         guard FileManager.default.fileExists(atPath: databaseURL.path) else {
-            return
+            return try Self.createDatabase(file: databaseURL, key: key, writerType: writerType, registerMigrationsHandler: registerMigrationsHandler)
         }
 
-        // create a new database file path
-        let newDbFile = self.nonExistingDBFile(withExtension: databaseURL.pathExtension, originalURL: databaseURL)
-
         // backup old db file
         let backupFile = self.nonExistingDBFile(withExtension: databaseURL.pathExtension + ".bak", originalURL: databaseURL)
         try FileManager.default.moveItem(at: databaseURL, to: backupFile)
 
-        // place just created new db in place of dbFile
-        try FileManager.default.moveItem(at: newDbFile, to: databaseURL)
```

Comment:
No new database file is created, causing 

`Error Domain=NSCocoaErrorDomain Code=4 "“Vault_1.db” couldn’t be moved to “DBP” because either the former doesn’t exist, or the folder containing the latter doesn’t exist." UserInfo={NSUnderlyingError=0x600003fc22e0 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory”}}`